rust_icu_uloc/src/lib.rs

```diff
@@ -385,6 +385,36 @@ impl ULoc {
         })
     }
 
+    /// Implements `uloc_openAvailableByType`.
+    pub fn open_available_by_type(locale_type: ULocAvailableType) -> Result<Enumeration, common::Error> {
+        let mut status = common::Error::OK_CODE;
+        unsafe {
+            let iter = Enumeration::from_raw_parts(None, versioned_function!(uloc_openAvailableByType)(locale_type, &mut status));
+            common::Error::ok_or_warning(status)?;
+            Ok(iter)
+        }
+    }
+
+    /// Returns a vector of locales of the requested type.
+    pub fn get_available_by_type(locale_type: ULocAvailableType) -> &'static Vec<ULoc> {
```

Comment:
It seems that more of this functionality will need to be pushed down to ICU.

Reply:
Are you okay with adding the static vectors to this library? The underlying assumption is that the ICU library is loaded once so that it is safe to cache these static results. If dynamically opening and closing the ICU library underneath this library is required then they may not be a good idea.

Reply:
Indeed, assuming static data is not going to work. ICU data can be loaded and unloaded.

In other places we've used this was to make it into iterator using Enumerate, but it seems that this particular API doesn't follow that pattern.

It might be OK to recompute the vector on every call, and have the user cache the results instead if they need to.

Reply:
Removed the caching and recomputing on every call - have a look.